packages/server/src/scripts/import_arpa_reporter_dump.js

```diff
@@ -0,0 +1,469 @@
+        .map((tenantId) => [
+            {
+                type: "input",
+                name: `tenantNames.${tenantId}`,
+                message: `Tenant name for source tenant ${tenantId}`,
+                validate: (name) =>
+                    name.trim().length > 0 || "must enter a name",
+            },
+            {
+                type: "search-list",
+                name: `mainAgencies.${tenantId}`,
+                message: `Main agency for source tenant ${tenantId}`,
+                choices: [
+                    {
+                        name: "(create new agency with same name as tenant)",
+                        value: null,
+                    },
+                    ...agencyOptionsByTenantId[tenantId].map((agency) => ({
+                        name: agency.name,
+                        value: agency.id,
+                    })),
+                ],
+            },
+        ])
+        .flatten()
+        .value();
+    let { tenantNames, mainAgencies } = await inquirer.prompt(questions);
+
+    // Annoying: by default Inquirer will make (sparse) arrays instead of objects; convert them back
+    // to the expected form.
+    tenantNames = _.fromPairs(
+        tenantNames
+            .filter((x) => x !== undefined)
+            .map((name, idx) => [idx, name])
+    );
+    mainAgencies = _.fromPairs(
+        mainAgencies
+            .filter((x) => x !== undefined)
+            .map((agencyId, idx) => [idx, agencyId])
+    );
+
+    // Create tenants
+    let inserted = [];
+    for (const tenantId of allTenantIds) {
+        const tenantName = tenantNames[tenantId];
+        const mainAgencyId = mainAgencies[tenantId];
+
+        const tenant = await knex("tenants")
+            .insert({
+                display_name: tenantName,
+                // main_agency_id must be populated in a second pass in importAgencies
+                // because there is a circular FK between tenants/agencies tables
+                main_agency_id: null,
+            })
+            .returning("*");
+        inserted.push(tenant);
+
+        if (mainAgencyId) {
+            tenant.__unmappedFutureMainAgencyId = mainAgencyId;
+        } else {
+            const agencyToInsert = {
+                id: 1000000 + tenantId,
+                tenant_id: tenantId,
+                name: tenantName,
+                code: `CHANGEME_${tenant.id}`,
+                __isMainAgency: true,
+            };
+            dbContents.agencies.splice(0, 0, agencyToInsert);
+            tenant.__unmappedFutureMainAgencyId = agencyToInsert.id;
+        }
+    }
+
+    const idLookup = _.chain(inserted)
+        .map((insertedRow, idx) => [allTenantIds[idx], insertedRow.id])
+        .fromPairs()
+        .value();
+
+    return { inserted, idLookup };
+}
+
+async function importAgencies(
+    dbContents,
+    idLookupByTable,
+    insertedRowsByTable
+) {
+    // First, create all agencies, defaulting their parent pointer to point to themselves.
```

Comment:
nit: This comment should mention both the parent pointer and the main_agency pointer